src/types/redis_bloom_chain.cc

```diff
@@ -190,17 +192,22 @@ rocksdb::Status BloomChain::InsertCommon(const Slice &user_key, const std::vecto
     } else {
       if (metadata.size + 1 > metadata.GetCapacity()) {
         if (metadata.IsScaling()) {
-          batch->Put(bf_key_list.back(), bf_data_list.back());
+          batch->Put(bf_key_list.back(), bf_data_list.back().ToStringView());
           std::string bf_data;
           createBloomFilterInBatch(ns_key, &metadata, batch, &bf_data);
-          bf_data_list.push_back(std::move(bf_data));
+          rocksdb::PinnableSlice pin_slice;
+          pin_slice.PinSelf(rocksdb::Slice(bf_data));
```

Comment:
Hmmm this part the logic is:

```
get bf_data ( a string ) in createBloomFilterInBatch
pin_slice.Pin(bf_data), which will clone the bf_data
```

Would you mind try:

```
*pin_slice.GetSelf() = std::move(bf_data);
```

I think it might be better than `PinSelf`.

Reply:
(This trick is used in `PinnableWideColumns::MoveValue` )

```
inline void PinnableWideColumns::MoveValue(std::string&& value) {
  std::string* const buf = value_.GetSelf();
  assert(buf);

  *buf = std::move(value);
  value_.PinSelf();
}
```

Reply:
Thanks for bring this up! I actually didn't realize this way to implementing it :)